### PR TITLE
KAFKA-5520: KIP-171 - Extend Consumer Group Reset Offset for Stream Application

### PR DESCRIPTION
**Pull request by jeqo**
KIP: https://cwiki.apache.org/confluence/display/KAFKA/KIP-171+-+Extend+Consumer+Group+Reset+Offset+for+Stream+Application

### CODE REVIEW

File: core/src/main/scala/kafka/tools/StreamsResetter.java [hunk 1]
```diff
@@ -89,71 +83,46 @@ public int run(final String[] args, final Properties config) {
...
+    private void parseArguments(final String[] args) {
 
         final OptionParser optionParser = new OptionParser(false);
         applicationIdOption = optionParser.accepts("application-id", "The Kafka Streams application ID (application.id).")
             .withRequiredArg()
             .ofType(String.class)
             .describedAs("id")
             .required();
-        bootstrapServerOption = optionParser.accepts("bootstrap-servers", "Comma-separated list of broker urls with format: HOST1:PORT1,HOST2:PORT2")
+        bootstrapServerOption = optionParser.accepts("bootstrap-server", "Comma-separated list of broker urls with format: HOST1:PORT1,HOST2:PORT2")
```
Comment (tedyu) [inline core/src/main/scala/kafka/tools/StreamsResetter.java]: Why removing the trailing 's' ?

Comment (jeqo) [reply core/src/main/scala/kafka/tools/StreamsResetter.java]: Forgot to mention it on the KIP, was thinking about aligning options. In Consumer Group command, and other, the option is `bootstrap-server`. Should I rollback this change?

Comment (tedyu) [reply core/src/main/scala/kafka/tools/StreamsResetter.java]: If it is not in KIP, better roll back.

Comment (jeqo) [reply core/src/main/scala/kafka/tools/StreamsResetter.java]: yeap, pushing changes.

**Comment by mjsax** [comment]
@jeqo Thanks for the PR. The KIP is not accepted yet, and there is no vote started. As a vote need to be open for 72h and KIPs must be accepted by tomorrow to be considered for 1.0 release, this will not make it anymore and we need to move it for 1.1 release.

\cc @guozhangwang Please correct me if I am wrong.

Thus, I would first try to get the KIP voted, and we can review this afterwards.

**Comment by guozhangwang** [comment]
There is a VOTE thread started already by @jeqo . I think implementation in parallel while waiting for the KIP to be accepted is ok.

**Comment by jeqo** [comment]
Closing this PR to merge with KIP-198: https://github.com/apache/kafka/pull/4159

**Comment by guozhangwang** [comment]
I'm not sure if we want to do these two KIPs in the same PR, but rather KIP-171 would depend on KIP-198 being done along which we will add the listTopics in the admin client. But that's just my understanding. @mjsax could you clarify?

**Comment by mjsax** [comment]
I agree. We should have 2 PR per KIP/JIRA.

**Comment by jeqo** [comment]
But KIP-198 is already merged https://cwiki.apache.org/confluence/display/KAFKA/KIP-198%3A+Remove+ZK+dependency+from+Streams+Reset+Tool :) 
What I mean is that I have updated the implementation according to KIP-198, that overlaps with KIP-171. I will update the issue title to avoid confusion.

**Pull request closed without merging**